### PR TITLE
Access code management guide

### PR DESCRIPTION
**Pull request by rchodava**

### CODE REVIEW

File: north-star-docs/access-codes.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+Receiving events requires that you have a webhook endpoint set up and registered. You can learn more about setting up a webhook and registering it in the [Webhooks](/core-concepts/webhooks.md) core concepts overview. Once you have a webhook registered (make sure to subscribe to the `access_code.*` events), you should receive events whenever an access code is created, updated, or deleted, or if there are any errors during the lifecycle. You can find descriptions of the various events, and the data they contain, in the [Events Reference](/api-clients/events/README.md).
+
+At a minimum, we recommend listening for the `access_code.set_on_device`, `access_code.removed_from_device`, `access_code.delay_in_setting_on_device` and `access_code.failed_to_set_on_device` events. Handling these events should serve as an alternative to polling for the status of an access code. Upon receiving an access code event, you should call the `GET /access_codes/get` endpoint to get the latest status, warnings and errors on an access code.
+
```
Comment (dawnho) [inline north-star-docs/access-codes.md:42]: What should a user do in response to receiving these events, other than just polling to get the latest status?

File: north-star-docs/access-codes.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+handle any errors that may occur.
+
+### Creating an access code
+
+Before creating an access code, you should check if the device supports access codes by inspecting the response from `/devices/get` or `/devices/list`.
+Supported devices will have `capabilities_supported.access_codes` set to `true`.
+
+Once you've confirmed that the device supports access codes:
+
+- Call the `POST /access_codes/create` endpoint with the appropriate parameters. Refer to the [API Reference](/api-clients/access-codes/create-an-access-code.md) for more details on the parameters you can pass to this endpoint. Also keep in mind that particular device providers may have constraints on access code pins and names. You can find more details on these constraints in the [Access Code Constraints Reference](/api-clients/devices##access-code-constraints) of the docs.
```
Comment (dawnho) [inline north-star-docs/access-codes.md:22]: The `create` endpoint often returns an action attempt. What should we expect from an action attempt? What do we do if the `code` is not returned?

Comment (dawnho) [reply north-star-docs/access-codes.md:22]: A lot of customers were confused about how to create access codes with the right timezone as well

File: north-star-docs/access-codes.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+- Use the `access_code_id` returned in the response to the `create` endpoint to call the `GET /access_codes/get` endpoint. A basic implementation would poll using this endpoint until the `status` of the access code is `set`. It's important to remember that for `time_bound` codes, the `status` will not be `set` until the `starts_at` time, so you should wait until that time to poll for the status.
+
+### Updating an access code
+
+An access code can be updated by calling the `POST /access_codes/update` endpoint with the appropriate parameters. Refer to the [API Reference](/api-clients/access-codes/update-an-access-code.md) for more details on the parameters you can pass to this endpoint.
+
+If you update an access code prior to the code getting set on a device, you can use the same polling mechanism described above to confirm that the code was successfully set. If you update an access code after the code has already been set on a device, you will need to listen for `access_code.*` events described in the Advanced Implementation section below to confirm that an access code is set.
```
Comment (dawnho) [inline north-star-docs/access-codes.md:29]: One bug / question that came up often during updates is customers would notice that the code would get removed from the lock. Is that the expected behavior? What happens when an active access code gets updated?

File: north-star-docs/access-codes.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
+# Access Codes
+
+## Overview
+
+The core concepts behind access codes are discussed in the [Access Codes](/core-concepts/access-codes.md) core concepts overview.
+We recommend reading the core concepts overview before reading this guide on implementing access codes.
+
```
Comment (dawnho) [inline north-star-docs/access-codes.md:7]: What happens when an access code starts, and what happens when an access code ends?

Do access codes get deleted automatically at the `ends_at` time?

(this was a question from a customer)

**Comment by rchodava** [comment]
@dawnho merging this in for now, if you do have further comments on changes, let me know and I can update in separate PR

**Pull request merged**